packages/api/src/car.js

```diff
@@ -85,23 +111,44 @@ export async function carPost (request, env) {
     local: blob.size > LOCAL_ADD_THRESHOLD
   })
 
+  // Retrieve current pin status and info about the nodes pinning the content.
+  const { peerMap } = await env.cluster.status(cid)
+
+  const pins = Object.entries(peerMap).map(([peerId, { peerName, status }]) => ({
+    status: toPinStatusEnum(status),
+    location: { peerId, peerName }
+  }))
+
+  if (!pins.length) { // should not happen
+    throw new Error('not pinning on any node')
+  }
+
   // Store in DB
-  await env.db.query(gql`
-    mutation importCar($data: ImportCarInput!) {
-      importCar(data: $data) {
-        name
-      }
-    }
-  `, {
+  const { importCar: upload } = await env.db.query(IMPORT_CAR, {
     data: {
       user: user._id,
       authToken: authToken?._id,
       cid,
-      name
+      name,
       // dagSize: undefined // TODO: should we default to chunk car behavior?
+      pins
     }
   })
 
+  // Partial CARs are chunked at ~10MB so anything less than this is probably complete.
+  if (ctx.waitUntil && upload.content.dagSize == null && blob.size < DAG_SIZE_CALC_LIMIT) {
+    ctx.waitUntil((async () => {
+      let dagSize
+      try {
+        dagSize = await getDagSize(blob)
+      } catch (err) {
+        console.error(`could not determine DAG size: ${err.stack}`)
+        return
+      }
+      await env.db.query(UPDATE_DAG_SIZE, { content: upload.content._id, dagSize })
+    })())
+  }
+
   // TODO: Improve response type with pin information
```

Comment:
wondering if this should receive the pin data?

Reply:
I think we should keep the API simple. They'll be able to request the status from `/status/:cid` to get that information immediately after.

Reply:
sounds good, so just get the commend removed